src/software/amazon/ion/IonLoader.java

```diff
@@ -144,7 +149,33 @@ public IonDatagram load(byte[] ionData)
      * @throws IonException if there's a syntax error in the Ion content.
      * @throws IOException if reading from the specified input stream results
      * in an <code>IOException</code>.
+     *
+     * @deprecated Will cause a memory leak when reading a gzipped stream.
+     * Use {@link IonLoader#load(IonReader)} instead.
      */
+    @Deprecated
     public IonDatagram load(InputStream ionData)
         throws IonException, IOException;
+
+
+    /**
+     * Loads an entire stream of Ion data into a single datagram,
+     * detecting whether it's text or binary data.
+     * <p>
+     * The specified reader remains open after this method returns.
+     * <p>
+     * This method will auto-detect and uncompress GZIPped Ion data.
```

Comment:
I'd suggest removing this line.  This method has nothing to do with whether the data is gzipped;  that information is solely related to the method used to construct the IonReader.